src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -29,6 +88,189 @@ public CommandBox(CommandExecutor commandExecutor) {
         this.commandExecutor = commandExecutor;
         // calls #setStyleToDefault() whenever there is a change to the text of the command box.
         commandTextField.textProperty().addListener((unused1, unused2, unused3) -> setStyleToDefault());
+
+        addKeyPressedEventForCommandTextField();
+        addKeyReleasedEventForCommandTextField();
+    }
+
+    private void addKeyPressedEventForCommandTextField() {
+        commandTextField.addEventFilter(KeyEvent.KEY_PRESSED, event -> {
+            if (event.getCode() == KeyCode.TAB) {
+                event.consume(); // Prevent the default tab behavior
+                if (commandTextField.getCaretPosition() == commandTextField.getText().length()) {
+                    autocomplete();
+                }
+            } else if (event.getCode() == KeyCode.SPACE) {
+                // Reset currentIndex and lastInput when space is pressed
+                currentIndex = -1;
+                lastInput = "";
+                System.out.println("[" + lastInput + "]");
+            }
+        });
+    }
+
```

Comment:
Could include JavaDoc here as well! 

Same goes for all the methods below.

Otherwise, it looks good to me! Well done :)